zerver/tests/test_message_send.py

```diff
@@ -2722,7 +2850,9 @@ def test_direct_message_to_self_and_bot_in_dm_disabled_org(self) -> None:
                 "content": "Test message",
             },
         )
-        self.assert_json_error(result, "Direct messages are disabled in this organization.")
+        self.assert_json_error(
+            result, "You are not allowed to send direct messages to these recipient(s)."
+        )
```

Comment:
If it's easy, I think it'd be worth preserving the legacy string for the special case where the `NOBODY` system group is what is selected for the setting.